### ISSUE TITLE
10

### ISSUE THREAD

**Issue by Gabrielonnn** [Gabrielonnn/IA#8]
import tensorflow as tf
from tensorflow.keras import Sequential
from tensorflow.keras.layers import Dense, Flatten
from tensorflow.keras.datasets import mnist
from tensorflow.keras.utils import to_categorical

def cargar_datos():
    (x_train, y_train), (x_test, y_test) = mnist.load_data()
    x_train, x_test = x_train / 255.0, x_test / 255.0
    y_train, y_test = to_categorical(y_train), to_categorical(y_test)
    return (x_train, y_train), (x_test, y_test)

def construir_modelo():
    modelo = Sequential([
        Flatten(input_shape=(28, 28)),
        Dense(128, activation='relu'),
        Dense(10, activation='softmax')
    ])
    modelo.compile(
        optimizer='adam',
        loss='categorical_crossentropy',
        metrics=['accuracy']
    )
    return modelo

def entrenar_modelo(modelo, x_train, y_train):
    modelo.fit(
        x_train, y_train,
        epochs=5,
        batch_size=32,
        validation_split=0.1
    )

def evaluar_modelo(modelo, x_test, y_test):
    loss, accuracy = modelo.evaluate(x_test, y_test)
    print(f"Precisión en test: {accuracy:.2f}")

def main():
    (x_train, y_train), (x_test, y_test) = cargar_datos()
    modelo = construir_modelo()
    entrenar_modelo(modelo, x_train, y_train)
    evaluar_modelo(modelo, x_test, y_test)

if __name__ == "__main__":
    main()
